src/dataflow/DataflowInferenceTreeAnnotator.java

```diff
@@ -105,13 +106,39 @@ public Void visitMethodInvocation(MethodInvocationTree methodInvocationTree,
                     .put(tm.toString(), tm);
             AnnotationMirror anno = DataflowUtils.genereateDataflowAnnoFromByteCode(atm,
                     this.realTypeFactory.getProcessingEnv());
+
+            if (atm instanceof AnnotatedArrayType) {
+                // If the return type of a byte code method is Array type,
+                // also generated Dataflow Annotation on Array component type.
+                relaceArrayComponentATM((AnnotatedArrayType) atm);
+            }
+
             replaceATM(atm, anno);
             return null;
         } else {
             return super.visitMethodInvocation(methodInvocationTree, atm);
         }
     }
 
+    /**
+     * Replace annotated type mirror by Dataflow Annotation for a given {@link AnnotatedArrayType}.<br>
```

Comment:
It's not clear from the first sentence what is replaced by what. Are you replacing something or adding the bytecode default?